src/gennylib/include/gennylib/conventions.hpp

```diff
@@ -124,12 +126,91 @@ inline bool operator==(const RateSpec& lhs, const RateSpec& rhs) {
     return (lhs.per == rhs.per) && (lhs.operations == rhs.operations);
 }
 
+struct PhaseRangeSpec {
+    PhaseRangeSpec() = default;
+    ~PhaseRangeSpec() = default;
+
+    PhaseRangeSpec(genny::IntegerSpec s, genny::IntegerSpec e) {
+        assert(s.value >= 0 && e.value >= 0 && s.value <= UINT_MAX && e.value <= UINT_MAX && s.value <= e.value);
+        start = static_cast<genny::PhaseNumber>(s.value);
+        end = static_cast<genny::PhaseNumber>(e.value);
+    }
+    PhaseRangeSpec(genny::IntegerSpec s) : PhaseRangeSpec(s, s) {}
+
+    // Allow construction with unsigned ints for testing
+    PhaseRangeSpec(u_int32_t s, u_int32_t e) : start{s}, end{e} {}
+    PhaseRangeSpec(u_int32_t s) : PhaseRangeSpec(s, s) {}
+
+    genny::PhaseNumber start;
+    genny::PhaseNumber end;
+};
+
 }  // namespace genny
 
 namespace YAML {
 
 using genny::decodeNodeInto;
 
+/**
+ * Convert between YAML and genny::PhaseRange
+ *
+ * The YAML syntax accepts "[genny::Integer]..[genny::Integer]".
+ * This is used to stipulate repeating a phase N number of times
+ */
+template <>
+struct convert<genny::PhaseRangeSpec> {
+    static Node encode(const genny::PhaseRangeSpec rhs) {
+        std::stringstream msg;
+        msg << rhs.start << ".." << rhs.end;
+        return Node{msg.str()};
+    }
+
+    static bool decode(const Node& node, genny::PhaseRangeSpec& rhs) {
+        if (node.IsSequence() || node.IsMap()) {
+            return false;
+        }
+        auto strRepr = node.as<std::string>();
+
+        // use '..' as delimiter.
+        const std::string delimiter = "..";
```

Comment:
`static` as well so it can be allocated in `.data`.

Reply:
I made it a constexpr.